code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -18,14 +18,15 @@
 	var/status = FALSE		//whether the thing is on or not
 	var/hitcost = 100
 	var/obj/item/weapon/cell/cell
-	var/spawn_cell = TRUE
+	var/obj/item/weapon/cell/starting_cell = /obj/item/weapon/cell/medium/high
```

Comment:
this new variable is unnecessary. you can work based on "cell" and suitable "cell".